flow/project.py

```diff
@@ -326,20 +337,99 @@ def cmd(self):
         try:
-            return JobStatus(self.job._project.document['_status'][self.id])
+            return JobStatus(self._jobs[0]._project.document['_status'][self.id])
         except KeyError:
             return JobStatus.unknown
 
 
 @deprecated(
     deprecated_in="0.11", removed_in="0.13", current_version=__version__)
 class JobOperation(_JobOperation):
-    pass
+    """This class represents the information needed to execute one group for one job.
+
+    The execution or submission of a :py:class:`FlowGroup` uses a passed-in command
+    which can either be a string or function with no arguments that returns a shell
+    executable command.  The shell executable command won't be used if it is
+    determined that the group can be executed without forking.
+
+    .. note::
+
+        This class is used by the :class:`~.FlowGroup` class for the execution and
+        submission process and should not be instantiated by users themselves.
+
+    :param id:
+        The id of this JobOperation instance. The id should be unique.
+    :type id:
+        str
+    :param name:
+        The name of the JobOperation.
+    :type name:
+        str
+    :param job:
+        The job associated with this operation.
+    :type job:
+        :class:`signac.contrib.job.Job`
+    :param cmd:
+        The command that executes this operation. Can be a callable that when
+        evaluated returns a string.
+    :type cmd:
+        callable or str
+    :param directives:
+        A dictionary of additional parameters that provide instructions on how
+        to execute this operation, e.g., specifically required resources.
+    :type directives:
+        :class:`dict`
+    """
+    def __init__(self, id, name, job, cmd, directives=None):
+        self._id = id
+        self.name = name
+        self._jobs = (job,)
+        if not (callable(cmd) or isinstance(cmd, str)):
+            raise ValueError("JobOperation cmd must be a callable or string.")
+        self._cmd = cmd
+
+        if directives is None:
+            directives = dict()  # default argument
+        else:
+            directives = dict(directives)  # explicit copy
+
+        # Keys which were explicitly set by the user, but are not evaluated by the
+        # template engine are cause for concern and might hint at a bug in the template
+        # script or ill-defined directives. We are therefore keeping track of all
+        # keys set by the user and check whether they have been evaluated by the template
+        # script engine later.
+        keys_set_by_user = set(directives)
+
+        # We use a special dictionary that allows us to track all keys that have been
+        # evaluated by the template engine and compare them to those explicitly set
+        # by the user. See also comment above.
+        self.directives = TrackGetItemDict(
+            {key: value for key, value in directives.items()})
+        self.directives._keys_set_by_user = keys_set_by_user
```

Comment:
```suggestion
        jobs = (job,)
        super().__init__(id, name, job, cmd, directives)
```

Reply:
After using a deprecated decorator, this class behaves like a method hence we have to manually define an init method

Reply:
I had forgotten thanks for the reminder.